### PR TITLE
Fix invalid conversions

### PR DESCRIPTION
**Pull request by victorapm**
This PR fixes several variable assignments that should be of type `HYPRE_BigInt`, but are currently `HYPRE_Int`.

### CODE REVIEW

**Review by liruipeng — APPROVED** [review]

**Review by rfalgout — APPROVED** [review]
Thanks, Victor!

**Comment by victorapm** [comment]
Thanks! @oseikuffuor1 I can hold this off if you need to merge `mp-build-all` first

**Comment by oseikuffuor1** [comment]
> Thanks! @oseikuffuor1 I can hold this off if you need to merge `mp-build-all` first

@victorapm There are a few more outstanding issues and discussions with the team before mp-build-all can be merged into master. If this is a bugfix PR, we shouldn't wait. I will review it as requested. Since it seems from the description that most changes are related to HYPRE_Int, there shouldn't be too many conflicts ... hopefully.

**Comment by victorapm** [comment]
sounds good! Yes, this is a bug fix only PR. I will merge then after getting all approvals

**Comment by victorapm** [comment]
I pushed two new bug fixes and I don't expect to push new changes, so this PR is ready for review

File: src/utilities/matrix_stats.c
Comment (ulrikeyang) [inline src/utilities/matrix_stats.c:145]: Why is coarse_num_rows BigInt, but fine_num_rows is not?

Comment (victorapm) [reply src/utilities/matrix_stats.c:145]: it has to be as well, thanks for catching!

File: src/parcsr_ls/par_nongalerkin.c
Comment (ulrikeyang) [inline src/parcsr_ls/par_nongalerkin.c:729]: row_start and row_stop should be HYPRE_Int  or  780, 781 and 785, 786 need to be cast to HYPRE_BigInt, since ijbuf_numcols is HYPRE_Int.

**Review by ulrikeyang — APPROVED** [review]
Other than the things I mentioned this looks fine.

File: src/parcsr_ls/par_nongalerkin.c
Comment (victorapm) [reply src/parcsr_ls/par_nongalerkin.c:729]: Let me update, thanks!

File: src/parcsr_ls/par_nongalerkin.c
Comment (oseikuffuor1) [inline src/parcsr_ls/par_nongalerkin.c:1689]: @victorapm why is the error checking removed from here and other places?

**Review by oseikuffuor1 — COMMENTED** [review]
@victorapm this looks good to me. I left one comment for you. Also, I didn't see where Ulrike's comments had been applied or resolved. Thanks.

File: src/parcsr_ls/par_nongalerkin.c
Comment (victorapm) [reply src/parcsr_ls/par_nongalerkin.c:1689]: If we want to access the current error code in hypre, we should use `HYPRE_GetError`. Therefore, we don't need to save error codes returned from HYPRE functions as done here

**Comment by victorapm** [comment]
> where Ulrike's comments had been applied or resolved

Sorry, they were in my local branch. Thanks!

**Review by oseikuffuor1 — APPROVED** [review]
Thanks!

**Pull request merged**
